src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,115 @@
+
+            StringBuilder textForSavingFile =
+                    generateReport(
+                            Integer.parseInt(processData[BUY_VALUE]),
+                            Integer.parseInt(processData[SUPPLY_VALUE]),textForSavingToFile);
+            writeToFile(textForSavingFile.toString(), writer);
+
+        } catch (IOException e) {
+            throw new NoSuchElementException(PARSE_EXCEPTION_MESSAGE);
+        }
+    }
+
+    private void writeToFile(String textForSavingFile, FileWriter writer) {
+        try {
+            writer.write(textForSavingFile);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private String[] processData(
+            String[] splitConstructTextFromFile,
+            int sumForSupply,
+            int sumForBuy) {
+        int copySumForSupply = sumForSupply;
+        int copySumForBuy = sumForBuy;
+        for (String s : splitConstructTextFromFile) {
+            String[] currentLine = s.split(SPLITERATOR);
+            switch (currentLine[KEY]) {
+                case CASE_BUY -> copySumForBuy += Integer.parseInt(currentLine[KEY_VALUE]);
+                case CASE_SUPPLY -> copySumForSupply += Integer.parseInt(currentLine[KEY_VALUE]);
+                default -> throw new NoSuchElementException(EXCEPTION_MESSAGE);
+            }
+        }
+        return new String[]{
+                CASE_SUPPLY,
+                String.valueOf(copySumForSupply),
+                CASE_BUY,
+                String.valueOf(copySumForBuy)
+        };
+    }
+
+    private StringBuilder generateReport(int sumForBuy, int sumForSupply,
+            StringBuilder textForSavingToFile) {
+        return textForSavingToFile.append(CASE_SUPPLY).append(SPLITERATOR)
+                .append(sumForSupply)
+                .append(System.lineSeparator())
+                .append(CASE_BUY)
+                .append(SPLITERATOR)
+                .append(sumForBuy)
+                .append(System.lineSeparator())
+                .append(RESULT)
+                .append(SPLITERATOR)
+                .append(sumForSupply - sumForBuy);
+    }
+
+    private String[] getSplitConstructTextFromFile(StringBuilder constructTextFromFile) {
+        return constructTextFromFile.toString().split(SPLIT_BY_NEW_LINE);
+    }
 
+    private void readFile(
+            BufferedReader reader,
+            StringBuilder constructTextFromFile) {
+        int value;
+        try {
+            value = reader.read();
+        } catch (IOException e) {
+            throw new RuntimeException(EXCEPTION_MESSAGE);
+        }
+        while (value != END_FILE_CODE) {
+            constructTextFromFile.append((char) value);
+            try {
+                value = reader.read();
+            } catch (IOException e) {
+                throw new RuntimeException(EXCEPTION_MESSAGE);
+            }
+        }
```

Comment:
Simplify this method, just read from file data and return it in String type, you don`t need 2 catch blocks here